MagicalCryptoWallet/Services/WalletService.cs

```diff
@@ -439,18 +439,47 @@ private void ProcessTransaction(SmartTransaction tx, List<HdPubKey> keys = null)
 			}
 		}
 
+		public class Operation
+		{
+			public Script Script { get; }
+			public Money Amount { get; }
+			public string Label { get; }
+
+			public Operation(Script script, Money amount, string label="")
+			{
+				Script = Guard.NotNull(nameof(script), script);
+				Amount = Guard.NotNull(nameof(amount), amount);
+				Label = label;
+			}
+		}
+
 		/// <param name="toSend">If Money.Zero then spends all available amount. Doesn't generate change.</param>
 		/// <param name="allowUnconfirmed">Allow to spend unconfirmed transactions, if necessary.</param>
 		/// <param name="allowedInputs">Only these inputs allowed to be used to build the transaction. The wallet must know the corresponding private keys.</param>
 		/// <param name="subtractFeeFromAmountIndex">If null, fee is substracted from the change. Otherwise it denotes the index in the toSend array.</param>
 		/// <exception cref="ArgumentException"></exception>
 		/// <exception cref="ArgumentNullException"></exception>
 		/// <exception cref="ArgumentOutOfRangeException"></exception>
-		public async Task<BuildTransactionResult> BuildTransactionAsync(string password, (Script script, Money amount, string label)[] toSend, int feeTarget, bool allowUnconfirmed = false, int? subtractFeeFromAmountIndex = null, Script customChange = null, IEnumerable<TxoRef> allowedInputs = null)
+		public async Task<BuildTransactionResult> BuildTransactionAsync(string password, Operation[] toSend, int feeTarget, bool allowUnconfirmed = false, int? subtractFeeFromAmountIndex = null, Script customChange = null, IEnumerable<TxoRef> allowedInputs = null)
 		{
-			password = password ?? ""; // Correction.
+			password = Guard.Correct(password);
```

Comment:
Please use the original one: `password = password ?? ""; // Correction.` `Guard.Correct` also trims, the password should not be trimmed.